SyslogNet.Client/Transport/SyslogLocalSender.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using SyslogNet.Client.Serialization;
+
+namespace SyslogNet.Client.Transport
+{
+	public class SyslogLocalSender : ISyslogMessageSender, IDisposable
+	{
+		// Will not work on Windows, as this relies on Unix system calls
+		public SyslogLocalSender()
+		{
+			PlatformID platform = Environment.OSVersion.Platform;
+			if (!(platform == PlatformID.MacOSX || platform == PlatformID.Unix)) {
+				throw new CommunicationsException("SyslogLocalSender is only available on Unix-like systems (e.g., Linux, BSD, OS X)");
+			}
+		}
+
+		[DllImport("libc", ExactSpelling=true)]

```

Comment:
Actually, I've given it quite a bit more thought, and there's a drawback to using Mono calls. I would have to reference the Mono.Posix assembly to have access to them, which naturally won't be found on most Windows machines. There are three options to get around this problem:

1. Change the build process for SyslogNet to have two different build targets, one for Windows and one for Mono. Then in the .csproj file, only add the reference when the build target is the Mono target. (That's how I've seen it done in other projects: they had a `MonoDebug` build target, and referenced Mono.Posix only when that build target was selected). **Drawback:** More complexity in building SyslogNet, as well as in distributing it: people have to download the correct DLL for the system they're using, and if they are trying to write a cross-platform app, they would have to include both DLLs and use the correct one on each platform. **My opinion:** :-1: due to extra complexity for both the project and its users.

1. Don't reference Mono.Posix in the assembly, but load it dynamically via reflection when a SyslogLocalTransport instance is created. **Drawback:** Reflection is slow, and every time a SyslogLocalTransport instance is created (which might be often), some amount of reflection will need to happen every time. Some things can be cached, but not everything. **My opinion:** :neutral_face: leaning towards :-1:. Would work, but the extra complexity isn't worth it, I feel. The benefit would be "if some hypothetical future platform shows up that has `syslog()` but not libc, we'd get ported to that platform "for free" when Mono is ported there." But I don't think that increased code complexity *now* is worth that hypothetical future benefit, especially as any such future platform would probably be very simple to port to. (Not to mention that Mono.Posix might not get ported to that hypothetical platform, since that porting effort might be immensely complex — so we'd have to write the port ourselves *anyway* and would have gained nothing from the increased complexity of SyslogNet code.)

1. Leave the code as-is, using `DllImport`s to get the `syslog()` *et al* functions. **Drawback:** Not necessarily portable to hypothetical future platforms. **My opinion:** Well, I'm biased since this would save me work, but :+1:.

Simplicity beats complexity most of the time, and beats *unnecessary* complexity 100% of the time. And the longer I think about the alternatives, the more I'm inclined to say "Just use `DllImport("libc")` and leave Mono out of it, so that we continue to work on Windows without needing reflection or a change to the build process."

If there's a good reason to use Mono.Posix that I haven't considered, I'm still willing to do the work. But for now, my recommendation is to use the approach that log4net took, and that I took in this PR, and keep it simple: just use `DllImport("libc")` and leave Mono.Posix out of it since it complicates matters on Windows.